twake/backend/node/src/services/console/web/controller.ts

```diff
@@ -34,15 +32,33 @@ export class ConsoleController {
     request: FastifyRequest<{ Body: AuthRequest }>,
     reply: FastifyReply,
   ): Promise<AuthResponse> {
-    if (request.body.access_token) {
-      return { access_token: await this.authByToken(request.body.access_token) };
+    if (request.body.remote_access_token) {
+      return { access_token: await this.authByToken(request.body.remote_access_token) };
     } else if (request.body.email && request.body.password) {
       return { access_token: await this.authByPassword(request.body.email, request.body.password) };
     } else {
-      throw CrudExeption.badRequest("access_token or email+password are required");
+      throw CrudExeption.badRequest("remote_access_token or email+password are required");
     }
   }
 
+  async tokenRenewal(request: FastifyRequest, reply: FastifyReply): Promise<AuthResponse> {
+    const authHeader = request.headers.authorization;
+    if (!authHeader || !authHeader.toLowerCase().startsWith("bearer")) {
+      throw CrudExeption.forbidden("Wrong authorization");
+    }
+
+    const token = authHeader.substr(7).trim();
+
+    let data: JWTObject = null;
+    try {
+      data = this.authService.verifyToken(token);
+    } catch (e) {
+      throw CrudExeption.forbidden(e.message);
+    }
```

Comment:
Checking token must be in the route preprocessing. There are places to do this with fastify, then it can be reused in other routes